warehouse/templates/includes/packaging/project-data.html

```diff
@@ -69,6 +70,7 @@ <h3 class="sidebar-section__title">{% trans %}Statistics{% endtrans %}</h3>
       </li>
     </ul>
   </div>
+  {% if not release.trusted_published %}</details>{% endif %}
```

Comment:
I think this is probably not worth doing this way, for a few reasons, and we should Do It Right™ instead.

As-is, we're not actually checking that the URL used for getting repo statistics is the same as the URL used for trusted publishing, so this can be subverted easily by using trusted publishing for something that is attempting to do "starjacking".

Additionally, it also doesn't indicate to the user or maintainer why sometimes statistics are hidden, and why sometimes they are not, which will make a more confusing UI.

I think we should remove this from the PR for now, with the plan to eventually replace it with a new statistics sidebar that does the following:

* if trusted publishing from GitHub has been used for the release:
  * include the URL of the upstream repository (https://github.com/pypi/warehouse/issues/10917)
  * fetch statistics for the repository that trusted publishing is configured for
  * mark the statistics as "verified via Trusted Publishing", with a link
* if trusted publishing was not used, and a GitHub URL was provided in the metadata (e.g., we would currently show statistics):
  * don't fetch the statistics
  * instead, include a message + provide a link that indicates that trusted publishing should be used to enable GitHub statistics

Reply:
@di Thanks for the feedback!

I've gone ahead and removed this change from scope so we can ship the other enhancements in the meantime.

I'll move the rest of the notes to a new issue so we can discuss the stats sidebar.